xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -1221,30 +1310,43 @@ struct AppEnvironment {
             gatewayURL: defaultGateway,
             sphereIdentity: defaultSphereIdentity
         )
+        self.noosphere = noosphere
 
         let databaseURL = self.applicationSupportURL
             .appendingPathComponent("database.sqlite")
 
-        let databaseService = DatabaseService(
+        let database = DatabaseService(
             database: SQLite3Database(
                 path: databaseURL.absoluteString,
                 mode: .readwrite
             ),
             migrations: Config.migrations
         )
+        self.database = database
         
-        let addresBook = AddressBookService(noosphere: noosphere, database: databaseService)
+        let addressBook = AddressBookService(
+            noosphere: noosphere,
+            database: database
+        )
+        self.addressBook = addressBook
         
         self.data = DataService(
             noosphere: noosphere,
-            database: databaseService,
+            database: database,
             local: local,
-            addressBook: addresBook
+            addressBook: addressBook
         )
         
-        self.addressBook = AddressBookEnvironment(data: data)
-
         self.feed = FeedService()
     }
 }
 
+extension AddressBookEnvironment {
```

Comment:
Extension to easily construct AddressBookEnvironment from AppEnvironment. Note relationship of extension = AppEnvironment knows about AddressBookEnvironment, but AddressBookEnvironment doesn't know about AppEnvironment. This mirrors component relationship where parents may know about children, but children may not know about parents.